lib/matplotlib/testing/decorators.py

```diff
@@ -41,23 +41,13 @@ def _knownfailureif(fail_condition, msg=None, known_exception_class=None):
     if the exception is an instance of this class. (Default = None)
 
     """
-    if is_called_from_pytest():
```

Comment:
This function is private, so we should probably just replace its callers with the direct pytest marker call.

Reply:
technically it is public but we should still do the switch and deprecate it.

Reply:
No it isn't. The one being removed below is public, but it's already deprecated. This one is private.

Reply:
Ah, you meant `is_called_from_pytest`; I meant the function in which this code exists `_knownfailureif`.